sklearn/cluster/tests/test_mean_shift.py

```diff
@@ -36,23 +36,28 @@ def test_estimate_bandwidth_1sample():
     # Test estimate_bandwidth when n_samples=1 and quantile<1, so that
     # n_neighbors is set to 1.
     bandwidth = estimate_bandwidth(X, n_samples=1, quantile=0.3)
-    assert_array_almost_equal(bandwidth, 0., decimal=5)
+    assert_equal(bandwidth, 0.)
 
 
-def test_mean_shift():
+@pytest.mark.parametrize("bandwidth, cluster_all, expected, "
+                         "first_cluster_label",
+                         [(1.2, True, 3, 0), (1.2, False, 4, -1)])
+def test_mean_shift(bandwidth, cluster_all, expected, first_cluster_label):
     # Test MeanShift algorithm
-    bandwidth = 1.2
-
-    ms = MeanShift(bandwidth=bandwidth)
+    ms = MeanShift(bandwidth=bandwidth, cluster_all=cluster_all)
     labels = ms.fit(X).labels_
     labels_unique = np.unique(labels)
     n_clusters_ = len(labels_unique)
-    assert_equal(n_clusters_, n_clusters)
+    assert n_clusters_ == expected
+    assert labels_unique[0] == first_cluster_label
 
-    cluster_centers, labels = mean_shift(X, bandwidth=bandwidth)
```

Comment:
Removing this means we are not testing the `mean_shift` function directly anymore.

Reply:
we are testing using
ms = MeanShift(bandwidth=bandwidth, cluster_all=cluster_all)
labels = ms.fit(X).labels_

Reply:
The testing of `mean_shift` should be independent of `ms.fit`. At the moment, `ms.fit` calls `mean_shift`, but we do not know how the code base will change.

Reply:
@thomasjpfan do we need another test for mean_shift?

Reply:
Leaving the original test here will sufficiently test `mean_shift`.

Reply:
@thomasjpfan added test for mean_shift as well